src/components/RegistrationForm.tsx

```diff
@@ -15,36 +20,45 @@ const RegistrationForm = ({isAuthorized, data }: AuthProps) => {
   }, [isAuthorized, router]);
 
   const handleSubmit = async (e: React.FormEvent) => {
-    e.preventDefault();
-
-    const entries = []
-    for (let i = 0; i < parseInt(periodLength as string); i++){
-      const newDate = new Date(startDate as string);
-      newDate.setDate(newDate.getDate() + (i + 1));
-      entries.push(postEntry(Router.asPath.includes('demo'), {
-        flow: "Medium",
-        craving: null,
-        mood: null,
-        symptom: null,
-        date: `${new Date(newDate).getFullYear()}-${new Date(newDate).getMonth() + 1}-${new Date(newDate).getDate()}`
-      }))
-      }
-      try {
-        await Promise.all(entries);
-        router.push("/dashboard");
-      }
-      catch (error) {
-        console.error(error);
-      }
+    if (periodLength < 1) {
+      e.preventDefault();
+      setError('Please enter a number equal to or greater than 1');
+      return;
     }
+    e.preventDefault();
+    setError('');
+    setSubmissionLoading(true);
+    await addUserAndEntries({name, birth_date: birthday, last_cycle_start: startDate, last_cycle_length: periodLength});
```

Comment:
```suggestion
    try { 
      await addUserAndEntries({name, birth_date: birthday, last_cycle_start: startDate, last_cycle_length: periodLength});
    } catch (error: any) {
      setError(`${error}`);
   }
```

What do you think about catching API errors and rendering them for the user? 💭